bootstrap/kubeadm/api/v1alpha4/kubeadmconfigtemplate_webhook_test.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilfeature "k8s.io/component-base/featuregate/testing"
+	bootstrapv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+	"sigs.k8s.io/cluster-api/feature"
+)
+
+// These tests are written in BDD-style using Ginkgo framework. Refer to
+// http://onsi.github.io/ginkgo to learn more.
+
+func Test_KubeadmConfigTemplate_validates_creation_and_deletion_and(t *testing.T) {
+	// NOTE: IgnitionBootstrapData feature flag is disabled by default. Enabling the feature flag
+	// temporarily for this test.
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.IgnitionBootstrapData, true)()
+
+	cases := map[string]struct {
+		in        *bootstrapv1.KubeadmConfigTemplate
+		expectErr bool
+	}{
+		"accepts valid configuration": {
```

Comment:
Not sure if that's the case with Ginkgo, but with `testing` package, spaces in test case names are converted to `_`, which makes them harder to grep for.
```suggestion
		"accepts_valid_configuration": {
```